src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/IProjectSnapshotExtensions.cs

```diff
@@ -40,7 +40,10 @@ public static RazorProjectInfo ToRazorProjectInfo(this IProjectSnapshot project,
 
     public static ImmutableArray<TagHelperDescriptor> GetTagHelpersSynchronously(this IProjectSnapshot projectSnapshot)
     {
-        Debug.Assert(projectSnapshot is ProjectSnapshot, "The ProjectSnapshot in the VisualStudioDocumentTracker should not be a cohosted project.");
+        var canResolveTagHelpersSynchronously = projectSnapshot is ProjectSnapshot ||
+            projectSnapshot.GetType().Name == "Microsoft.VisualStudio.Editor.Razor.EphemeralProjectSnapshot";
```

Comment:
>EphemeralProjectSnapshot [](http://example.com/codeflow?start=83&length=24)

For my education, what's EphemeralProjectSnapshot? :)

Reply:
You tell me, and we'll both know!

No, okay, its a project snapshot that is returned when one of our VS editor components needs to have a project, but we haven't been told about any projects yet (eg, you're opening VS and a .razor file is already open from last session) or possibly use for loose files that aren't in any project.

Reply:
So basically similar to Misc Files project?

Reply:
Yeah, though it isn't as long lived as that. Eg, if you ask for the project twice, I'm pretty sure you'll get two different instances. The language server has a proper misc files project, no idea why the VS side doesn't.

Reply:
Yeah, there's only one place in our code that actually constructs an `EphemeralProjectSnapshot`. It's something that I'd like to get rid of.